styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java

```diff
@@ -168,6 +186,43 @@ private BackfillsPayload getBackfills(RequestContext requestContext) {
     return Response.ok();
   }
 
+  private void haltActiveBackfillInstances(Backfill backfill, Client client) throws IOException {
+    final List<RunStateData> activeRunStateData = retrieveBackfillStatuses(backfill).stream()
+        .filter(this::activeState).collect(toList());
+    if (activeRunStateData.size() != 0) {
+      final boolean failed =
+          activeRunStateData.parallelStream().map(RunStateData::workflowInstance)
+              .map(workflowInstance -> haltActiveBackfillInstance(workflowInstance, client))
+              .anyMatch(success -> !success);
+      if (failed) {
+        // TODO: shall we use customized exception?
+        throw new IOException("some active workflow instances cannot be halted");
+      }
+    }
+  }
+
+  private boolean haltActiveBackfillInstance(WorkflowInstance workflowInstance, Client client) {
+    try {
+      final ByteString payload = serialize(Event.halt(workflowInstance));
+      final Request request = Request.forUri(schedulerApiUrl("events"), "POST")
+          .withPayload(payload);
+      return client.send(request)
+          .thenApply(response -> response.status().family().equals(SUCCESSFUL))
+          .toCompletableFuture().get();
```

Comment:
A little bit complicated in this case. It will become a CompletableFuture of a CompletableFuture, and exception handling becomes hard to read.